src/depthquality/quality.py

```diff
@@ -87,6 +91,15 @@ def clip_pointcloud_to_pattern_area(reference_mesh, aligned_pointcloud, depth_sc
 
     return cropped_pointcloud
 
+def compute_corner_coordinates(pointcloud, camera_matrix, corner_coordinates):
```

Comment:
It's always scary to pass in a dictionary that will be mutated, because python is scary and lets you do that (with no way to specify pass-by-reference and pass-by-value). It would be a bit more readable if this function took in a list of `(u, v)` coordinates and constructed the dictionary of `(u, v) -> (x, y, z)` as the return output of this function. You're not incrementally calling this function on top of `corner_coordinates` multiple times so there isn't utility in mutating the dictionary in place.

Reply:
Thanks. That makes total sense. Implemented this in the latest commit.